packages/travix-ui-kit/tests/unit/stepper/stepper.render.spec.js

```diff
@@ -0,0 +1,40 @@
+import { shallow } from 'enzyme';
+import React from 'react';
+import Stepper from '../../../components/stepper/stepper';
+
+describe('Stepper', () => {
```

Comment:
Let's not test React lifecycle methods.

Reply:
I'm testing snapshots here. I'm sure we need to test it

Reply:
That's test for snapshots, I'm sure we need to keep it here

Reply:
Why not simplify it to a single file that tests the render, callback after click on increase and callback after click on decrease?

Reply:
I'm sure it's not a way of unit testing components
Also I've checked tests for other components and tests are written in same way.

Reply:
You're testing if some specifics methods are being called when your method is called.

The great thing with Jest + Enzyme is that you don't have to test specific internals methods of your components. You can mount/shallow them and test if it's rendering correctly with different props being passed or if after some user interaction your component renders correctly and your callbacks/side effects are being called with the right data.

Otherwise, you will have to update your tests after a refactor that should not change the way how your component behaves.  

What I was expecting:
```
~/stepper.spec.js

describe Stepper
it should render with both buttons enabled (initValue = 1, minValue = 0, maxValue = 2)
it should render with the decrement button disabled (initValue = 0, minValue = 0, maxValue = 2)
it should render with the increment button disabled  (initValue = 2, minValue = 0, maxValue = 2)
it should disable increment button if reaches the maximum value
it should enable increment button if it's not the maximum value anymore
it should disable decrement button if reaches the minimum value
it should enable decrement button if it's not the minimum value anymore
it should not allow decrementing less than the minimum value
it should not allow incrementing more than the maximum value
it should call onChange callback with the correct value after a change
it should not call onChange callback when the minimum value was reach
it should not call onChange callback when the maximum value was reach
```

If you test this with Jest + Enzyme (mount, shallow) you don't need to test any internal method and the test will not be attached to your implementation but to the component desired behavior.

Reply:
I got what you mean, but why do not we have any other components, tested in the same way?
And again, you advice to write not unit tests.

Reply:
Guys, I think Snapshots testing is very cool and powerful feature, which helps us to test the presentation of React components.

Reply:
In my point of view, snapshots should work together with unit test. Because snapshot testing is some kind of Integration tests, but not unit.
Nice article about it: https://hackernoon.com/front-end-react-snapshot-testing-with-jest-what-is-it-for-7788f7bd5a2e